src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PhysicalFilesWatcher.cs

```diff
@@ -165,12 +165,17 @@ private IChangeToken GetOrAddChangeToken(string pattern)
             }
             else
             {
-                changeToken = GetOrAddFilePathChangeToken(pattern);
+                // get rid of \. in Windows and ./ in UNIX's at the start of path file
+                var filePath = RemoveRelativePathSegment(pattern);
```

Comment:
Is this a breaking change?

Reply:
I don't think it breaks anything, it looks like a bug fix to me. Unless, there's someone relying on the fact that on change does not get fired for his files when specifying them with starting ./ which is highly improbable because he could just pass false to AddJsonFile